cmd/check_container.go

```diff
@@ -152,111 +152,117 @@ var checkContainerCmd = &cobra.Command{
+			err = json.Unmarshal(rpmManifestBytes, &rpmManifest)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"rpm manifest",
+					err,
+				)
 			}
 
-			certImage := new(pyxis.CertImage)
-			err = json.Unmarshal(certImageBytes, &certImage)
+			// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
+			var testResults pyxis.TestResults
+			err = json.Unmarshal(formattedResults, &testResults)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"preflight results.json",
+					err,
+				)
 			}
 
-			certImage.ISVPID = certProject.Container.ISVPID
+			// Populate the certImage with missing values as a result of this execution.
+			certImage.ISVPID = certProject.Container.ISVPID // NOTE: certProject depends on Pyxis calls made outside this block.
 			certImage.Certified = testResults.Passed
 
-			rpmManifestJsonFile, err := os.Open(path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename))
-			if err != nil {
-				return err
-			}
-			defer rpmManifestJsonFile.Close()
-
-			rpmManifestBytes, err := io.ReadAll(rpmManifestJsonFile)
-			if err != nil {
-				return err
-			}
-
-			rpmManifest := new(pyxis.RPMManifest)
-			err = json.Unmarshal(rpmManifestBytes, rpmManifest)
-			if err != nil {
-				return err
-			}
-
+			// Send the preflight logfile as an artifact in our submission.
 			logFileName := viper.GetString("logfile")
-
-			logFile, err := os.Open(logFileName)
-			if err != nil {
-				return err
-			}
-			defer logFile.Close()
-
-			logFileBytes, err := io.ReadAll(logFile)
+			logFileBytes, logFileSize, err := readFileAndGetSize(logFileName)
 			if err != nil {
-				return err
-			}
-
-			logFileInfo, err := logFile.Stat()
-			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: unable to read file from disk to include in submission %s: %s",
+					errors.ErrSubmittingToPyxis,
+					logFileName,
+					err,
+				)
 			}
 
 			logFileArtifact := pyxis.Artifact{
 				CertProject: projectId,
 				Content:     base64.StdEncoding.EncodeToString(logFileBytes),
 				ContentType: http.DetectContentType(logFileBytes),
 				Filename:    logFileName,
-				FileSize:    logFileInfo.Size(),
+				FileSize:    logFileSize,
 			}
 
-			artifacts := make([]pyxis.Artifact, 0, 1)
-			artifacts = append(artifacts, logFileArtifact)
-
+			// assemble all of the components into the expected POST input format.
+			// and submit the results.
 			certInput := &pyxis.CertificationInput{
 				CertProject: certProject,
-				CertImage:   certImage,
-				RpmManifest: rpmManifest,
-				TestResults: testResults,
-				Artifacts:   artifacts,
+				CertImage:   &certImage,
+				RpmManifest: &rpmManifest,
+				TestResults: &testResults,
+				Artifacts:   []pyxis.Artifact{logFileArtifact},
```

Comment:
Not sure about this change. There will be more artifacts (In fact, I think we should already be sending up the files that we've put on disk, for comparison purposes), so I think having this defined as above should remain.

Reply:
I agree with @bcrochet on this, a list was intended for us to send more files on disk as they are needed.

Reply:
Gotcha. We were instantiating an empty slice and then appending logFileArtifact, which we already had at the time (and could have just initialized with the value already in place). I can revert this.

Reply:
I reverted this for the moment, though I still instantiate the slice with our existing value and avoid an extra resize operation (append)

Reply:
No strong feeling there. But leaving the creation of the slice outside the creation of the results is ideal.